yourapp/your_workflow_definition_replay_test.go

```diff
@@ -0,0 +1,44 @@
+package yourapp
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"go.temporal.io/api/workflowservicemock/v1"
+	"go.temporal.io/sdk/worker"
+)
+
+type replayTestSuite struct {
+	suite.Suite
+	mockCtrl *gomock.Controller
+	service  *workflowservicemock.MockWorkflowServiceClient
+}
+
+func TestReplayTestSuite(t *testing.T) {
+	s := new(replayTestSuite)
+	suite.Run(t, s)
+}
+
+func (s *replayTestSuite) SetupTest() {
+	s.mockCtrl = gomock.NewController(s.T())
+	s.service = workflowservicemock.NewMockWorkflowServiceClient(s.mockCtrl)
+}
+
+func (s *replayTestSuite) TearDownTest() {
+	s.mockCtrl.Finish() // assert mock’s expectations
+}
+
+// This replay test is the recommended way to make sure changing workflow code is backward compatible without non-deterministic errors.
+// "yourworkflow.json" can be downloaded from Temporal CLI:
+//      tctl wf show -w your-workflow-id --output_filename ./your-workflow-id.json
+// Or from Temporal Web UI. And you may need to change workflowType in the first event.
+func (s *replayTestSuite) TestReplayWorkflowHistoryFromFile() {
+	replayer := worker.NewWorkflowReplayer()
+
+	replayer.RegisterWorkflow(YourWorkflowDefinition)
+
+	err := replayer.ReplayWorkflowHistoryFromJSONFile(nil, "yourworkflow.json")
```

Comment:
If tests are run from another directory, this may fail. You can `go:embed` or you can do something like `_, currFile, _, _ := runtime.Caller(0)` + `filepath.Dir(currFile)`

Reply:
I tested your statement and ran the tests from a bunch of other directories without issue - so... this seems to be just fine?

Reply:
Yes, it seems it sets the local directory. But usually users shouldn't count on local directory on their runs. Yes this is fine, but if users copy this code for use outside of tests (which many do for replayers) it'll break.